packages/supermassive/hooks.md

```diff
@@ -0,0 +1,64 @@
+# Hooks Documentation
+
+## Overview
+
+This document describes the behaviour of hooks when they encounter errors.
+
+## General Rule
+
+- **Returned Errors**: The error is registered and execution continues.
+- **Thrown Errors**: Specific behaviour is applied based on the hook.
+
+### Hooks and Their Behaviours
+
+#### `beforeOperationExecute`
+
+Called before every operation
+
+- **Thrown Error**: Stops execution and sets `data` to `null` and registers the error.
```

Comment:
Make all those be in same order, sometimes youh ave returned first, sometimse thrown first.